Plugins/GitUIPluginInterfaces/ISubmodulesConfigFile.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace GitUIPluginInterfaces
+{
+    public interface ISubmodulesConfigFile
+    {
+        IReadOnlyList<IConfigSection> ConfigSections { get; }
+
+        string GetPathValue(string setting);
+
+        void RemoveConfigSection(string configSectionName);
+
+        void Save();
+    }
```

Comment:
Rather extend `IConfigFileSettings`:

https://github.com/gitextensions/gitextensions/blob/4eae18437d306aeb8ca676b78b58a1494c5db8b0/Plugins/GitUIPluginInterfaces/IConfigFileSettings.cs#L3-L27

Reply:
I tried that, and that'd be rather incoherent from the implementation standpoint (the whole current "config"-related set of implementations is incoherent, but that'd just make it worse).

`ISubmodulesConfigFile` is only needed for `ConfigFile`, whereas `IConfigFileSettings` is implemented by `ConfigFileSettings`, which is used in completely different paths.

I extracted the interface purely for the purpose of removing the dependency on the concrete implementation and getting the code to compile. I picked the name purely because it denoted the type returned by `GetSubmoduleConfigFile()` call.

Please feel free to pull down the branch and propose a better alternative.